drivers/serial/uart_nrfx_uarte.c

```diff
@@ -485,6 +483,12 @@ static int uarte_nrfx_tx(struct device *dev, const u8_t *buf, size_t len,
 	data->async->tx_buf = buf;
```

Comment:
There is still a chance of incorrect behavior here. 
If this function is pre-empted after the check above, but before these 2 lines run, the check is no longer valid.
Thread 2 could have started a TX, this function returns and overwrites data->async.
The original TX will never have the user callback called on.

Reply:
yep, for that i would recommend `atomic_cas` something like:
`if (atomic_cas(&txbuf, NULL, buf)==false) return -EBUSY;`